pypeit/core/extract.py

```diff
@@ -1850,19 +1900,35 @@ def ech_objfind(image, ivar, slitmask, slit_left, slit_righ, order_vec, maskslit
     gdorders = np.arange(norders)[np.invert(maskslits)]
     for iord in gdorders: #range(norders):
         msgs.info('Finding objects on order # {:d}'.format(order_vec[iord]))
-        thismask = slitmask == gdslit_spat[iord]
-        inmask_iord = inmask & thismask
+        gpmmask = slitmask == gdslit_spat[iord]
+        inmask_iord = inmask & gpmmask
         specobj_dict['SLITID'] = iord
         specobj_dict['ECH_ORDERINDX'] = iord
         specobj_dict['ECH_ORDER'] = order_vec[iord]
         try:
             std_in = std_trace[:,iord]
         except TypeError:
             std_in = None
-        sobjs_slit, skymask_objfind[thismask] = \
-            objfind(image, thismask, slit_left[:,iord], slit_righ[:,iord], spec_min_max=spec_min_max[:,iord],
+
```

Comment:
The way this code works, you should only need to create a single hand object, not one at every location on the orders. Touch base with me about how to implement this correctly.

Reply:
TODO ADDED.